samples/browser/src/main/java/org/mozilla/samples/browser/DefaultComponents.kt

```diff
@@ -132,7 +137,7 @@ open class DefaultComponents(private val applicationContext: Context) {
 
     // Addons
     val addonManager by lazy {
-        AddonManager(store, addonCollectionProvider)
+        AddonManager(store, engine, addonCollectionProvider, addonUpdater)
```

Comment:
Hmm now we have `AddonManager` depends on `DefaultAddonUpdater` and `AddonCollectionProvider`. I wonder if we could have one consumer component declaration instead of needing to declare a multiple components (as it appears as of now most UI integrations are done through `AddonManager`). I'm curious to see if the simplicity of client side integration is worth looking into (since declaring multiple components can be confusing). Thoughts?

Reply:
Yes, so it's definitely worth looking into at some point, but seeing our current component graph I don't think we need to do this as part of this PR: https://github.com/mozilla-mobile/android-components/blob/master/samples/browser/src/main/java/org/mozilla/samples/browser/DefaultComponents.kt

Like there's lots than can be optimized. In reference-browser, this looks much cleaner as we created component groups by that also has disadvantages (all components in a group are initialized together which is tricky for startup performance).

Reply:
I agree with optimization being out of this PR